src/routes/view-content/[guideId]/[bibleSection]/library-resource-menu/SearchByResource.svelte

```diff
@@ -0,0 +1,145 @@
+<script lang="ts">
+    import { _ as translate } from 'svelte-i18n';
+    import XMarkIcon from '$lib/icons/XMarkIcon.svelte';
+    import { parentResources } from '$lib/stores/parent-resource.store';
+    import type { ApiParentResource } from '$lib/types/resource';
+    import SearchInput from '$lib/components/SearchInput.svelte';
+    import { getContentContext } from '../context';
+    import type { ResourceContentInfo, ResourceContentInfoWithMetadata } from '$lib/types/resource';
+    import FullPageSpinner from '$lib/components/FullPageSpinner.svelte';
+    import ChevronLeftIcon from '$lib/icons/ChevronLeftIcon.svelte';
+
+    export let resourceSearchResources: ResourceContentInfo[];
+    export let searchByParentResource: (
+        parentResource: ApiParentResource | null,
+        offset: number | null,
+        query: string
+    ) => Promise<void>;
+    export let resourceSelected: (resource: ResourceContentInfoWithMetadata) => void;
+    export let isLoading: boolean;
+
+    const { isResourceSearch, setIsResourceSearch } = getContentContext();
+
+    let showParentResourceMenu = true;
+    let showResourceMenu = false;
+    let currentParentResource: ApiParentResource | null;
+    let searchQuery = '';
+
+    $: searchByParentResource(currentParentResource, 0, searchQuery);
+
+    function closeParentResourceMenu() {
+        setIsResourceSearch(false);
+        showParentResourceMenu = false;
+    }
+
+    function closeResourcesMenu() {
+        resourceSearchResources = [];
+        showResourceMenu = false;
+        setIsResourceSearch(false);
+        searchQuery = '';
+    }
+
+    function backToParentResourceMenu() {
+        resourceSearchResources = [];
+        searchQuery = '';
+        showResourceMenu = false;
+        showParentResourceMenu = true;
+    }
+
+    async function parentResourceSearch(parentResource: ApiParentResource) {
+        showResourceMenu = true;
+        showParentResourceMenu = false;
+        currentParentResource = parentResource;
+        await searchByParentResource(parentResource, 0, searchQuery);
+    }
+
+    async function loadMoreResources() {
+        if (currentParentResource) {
+            await searchByParentResource(currentParentResource, resourceSearchResources.length, searchQuery);
+        }
+    }
+</script>
+
+{#if $isResourceSearch}
```

Comment:
Nit: we already check `isResourceSearch` before rendering this so could remove this here and unindent the rest of it?